File: 242-valid-anagram/valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        map<char, int> ak;
        map<char, int> kb;
        for(char c : s){
            ak[c]++;
        }
        for(char c : t){
            kb[c]++;
        }
        for(char c : s){
            if(ak[c] != kb[c]) return false;
        }
        for(char c : t){
            if(ak[c] != kb[c]) return false;
        }
        return true;
    }
};

File: 242-valid-anagram/valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        Map<Character, Integer> ak = new HashMap<>();
        Map<Character, Integer> kb = new HashMap<>();
        if(s.length() != t.length()) return false;
        for(char c : s.toCharArray()){
            ak.put(c,ak.getOrDefault(c, 0) + 1);
        }
        for(char c : t.toCharArray()){
            kb.put(c,kb.getOrDefault(c, 0) + 1);
        }
        return ak.equals(kb);
    }
}